pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -381,6 +377,74 @@ public synchronized TaskState getTaskState(String taskName) {
     return filteredTaskStateMap;
   }
 
+  /**
+   * Given a taskType, helper method to debug all the HelixJobs for the taskType.
+   * For each of the HelixJobs, collects status of the (sub)tasks in the taskbatch.
+   *
+   * @param taskType      Pinot taskType / Helix JobQueue
+   * @param verboseLevel  By default, does not show details for completed tasks.
+   *                      If verboseLevel >= 10, shows details for all tasks.
+   * @return Map of Pinot Task Name to TaskDebugInfo. TaskDebugInfo contains details for subtasks.
+   */
+  public synchronized Map<String, TaskDebugInfo> getTaskDebugInfo(String taskType, int verboseLevel) {
+    Map<String, TaskDebugInfo> taskDebugInfos = new HashMap<>();
+    WorkflowContext workflowContext = _taskDriver.getWorkflowContext(getHelixJobQueueName(taskType));
+    if (workflowContext == null) {
+      return taskDebugInfos;
+    }
+    boolean showCompleted = verboseLevel >= 10;
+    long timeMillis;
+    SimpleDateFormat SIMPLE_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z", Locale.getDefault());
+    SIMPLE_DATE_FORMAT.setTimeZone(TimeZone.getDefault());
+    Map<String, TaskState> helixJobStates = workflowContext.getJobStates();
+    for (String helixJobName : helixJobStates.keySet()) {
+      String pinotTaskName = getPinotTaskName(helixJobName);
+      TaskDebugInfo taskDebugInfo = new TaskDebugInfo();
+      taskDebugInfo.setTaskState(workflowContext.getJobState(helixJobName));
+      timeMillis = workflowContext.getJobStartTime(helixJobName);
+      taskDebugInfo.setStartTime((timeMillis <= 0) ? null : SIMPLE_DATE_FORMAT.format(timeMillis));
```

Comment:
(nit) for readability. Same for other time handling
```suggestion
      long jobStartTimeMs = workflowContext.getJobStartTime(helixJobName);
      if (jobStartTimeMs > 0) {
        taskDebugInfo.setStartTime(SIMPLE_DATE_FORMAT.format(jobStartTimeMs));
      }
```

Reply:
cleaned